modules/gmc-agent/src/main/java/org/gridgain/agent/Agent.java

```diff
@@ -103,12 +111,14 @@ public Agent(GridKernalContext ctx) {
     /** {@inheritDoc} */
     @Override public void onKernalStart(boolean active) {
         spanExporter = new GmcSpanExporter(ctx);
+        nodeConfigurationExporter = new NodeConfigurationExporter(ctx);
         metaStorage = ctx.cache().context().database().metaStorage();
 
         launchAgentListener(null, ctx.discovery().discoCache());
 
         // Listener for coordinator changed.
         ctx.event().addDiscoveryEventListener(this::launchAgentListener, EVTS_DISCOVERY);
+        nodeConfigurationExporter.export();
```

Comment:
I think it is not good idea to do potentially "heavy" operations in onKernalStart method. 
May be backend should "poll" for configurations from cluster?

Reply:
I don't think what putting element in queue is "heavy" operation (yes, we are wrapping configuration but in the future we will have a serializable IgniteConfiguration). Maybe we can suggest an additional type of callbacks (afterStart for example) to core team?